### DISCUSSION TITLE
How to crop 2D array by dropping nan values outside

### DISCUSSION THREAD

**Discussion by zxdawn** [pydata/xarray#8798] [Q&A]
Hi, I'm trying to drop nan values of 2D Array and plot the image. However, I wanna keep the original distribution with nan values inside.

Here's an example:
```
import numpy as np
import xarray as xr
import matplotlib.pyplot as plt
import matplotlib.patches as patches

data = np.array([[np.nan, np.nan, np.nan, np.nan],
                 [np.nan, np.nan, np.nan, 3],
                 [1, np.nan, np.nan, 4],
                 [2, np.nan, np.nan, 5],
                 [np.nan, np.nan, np.nan, 6],])

da_xarray = xr.DataArray(data, dims=['y', 'x'])

fig, ax = plt.subplots()
da_xarray.plot()

# preferred image region
rect = patches.Rectangle((-0.5, 0.5), 4, 4, linewidth=3, edgecolor='r', facecolor='none')
ax.add_patch(rect)
```

I wanna plot the Array inside the red box.
![image](https://github.com/pydata/xarray/assets/30388627/e3318c59-5828-4554-a623-fbac4f0c6492)

I have tried to use `dropna`:
```
da_xarray.dropna('y', how='all').dropna('x', how='all').plot()
```

This will drop nan values between them.
![image](https://github.com/pydata/xarray/assets/30388627/f8a235fd-8655-497e-8761-325270f133e2)

Any idea how to plot valid data and keep the red box part?

Because the nan values are random in my real data, I can not use `sel` or `isel` to select them manually.

**Comment by dcherian** [comment]
`da_xarray.where(da_xarray.notnull(), drop=True)`?

**Reply by zxdawn** [reply]
Thanks for the suggestion. It gives the wrong result as same as applying `drop` on each axis twice:

![image](https://github.com/pydata/xarray/assets/30388627/28d2ef21-fe7d-4d5a-95df-59f798ad77b8)

I wanna get this result:

![image](https://github.com/pydata/xarray/assets/30388627/47cc78e0-c57c-4b2e-856b-986d4689e1ab)

**Reply by dcherian** [reply]
oh right, yeah I think you'll have to do it manually.
